WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -41,7 +41,7 @@ public CoinJoinManager(IWalletProvider walletProvider, RoundStateUpdater roundSt
 	public RoundStateUpdater RoundStatusUpdater { get; }
 	public string CoordinatorIdentifier { get; }
 	private CoinRefrigerator CoinRefrigerator { get; } = new();
-	public bool IsUserInSendWorkflow { get; set; }
+	private HashSet<string> WalletsInSendWorkflow { get; } = new ();
```

Comment:
This is observed and manipulated by multiple threads. Use `ConcurrentDictionary<TKey, TValue>` - safer.

Reply:
I am not sure about `Dictionary`, having a `value` doesn't really make sense.

Taking a look at the [Thread-Safe Collections](https://learn.microsoft.com/en-us/dotnet/standard/collections/thread-safe/) we have two options:
1. [ConcurrentBag<T>](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentbag-1)
   - The disadvantage is that it allows duplications. But it can be handled during adding.
2. `ConcurrentDictionary`
   - And store some dummy (not used) values. No duplications, thread-safe, but does not perfectly suit this task.

Reply:
ConcurrentBag cannot be used for this scenario as a specific item can not be taken out of it, that is why it's a bag.
I implemented it with `ConcurrentDictionary` with an added comment about the Value is not used, only the key.

If there are more scenarios like this, we could add our own thread-safe list, I have seen some examples.